docs/fundamentals/code-analysis/style-rules/index.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Code style rules overview
+ms.date: 09/25/2020
+ms.topic: reference
+author: gewarren
+ms.author: gewarren
+---
+# Code style rules
+
+.NET code style analysis provides rules that aim to maintain consistent *code style* in your codebase. These rules have an "IDE" prefix in the rule ID. Majority of the rules have an associated code style option to customize the preferred style. The rules are organized into following sub-categories:
+
+- [Language rules](language-rules.md)
+
+   Rules pertaining to the C# or Visual Basic language. For example, you can specify rules around using `var` or explicit types when defining variables, or preferring expression-bodied members.
+
+- [Formatting rules](formatting-rules.md)
+
+   Rules regarding the layout and structure of your code in order to make it easier to read. For example, you can specify rules around Allman braces, or preferring spaces in control blocks.
+
+- [Naming rules](naming-rules.md)
+
+   Rules regarding the naming of code elements. For example, you can specify that `async` methods must end in "Async".
+
+## Legend
```

Comment:
We will eventually have an `Index` section above this legend, similar to the Code quality rule index page. This will be populated once we start creating dedicated per-ID or per-option documentation pages in follow-up PRs.